src/Streamly/Internal/Data/Stream/Concurrent.hs

```diff
@@ -62,22 +62,32 @@ module Streamly.Internal.Data.Stream.Concurrent
     -- ** Combine two
     -- | Use a channel for each pair.
     -- combine/concur/conjoin
+    , append2
+    , interleave2
+    , ahead2
+    , parallel2
+    , parallelFst2
+    , parallelMin2
+    , combineWith
```

Comment:
We should encourage the use of combining many streams at once or explicitly use channels to combine streams. Even if they combine 2 streams they can always keep the channel around to send another stream to it later. 

So, first use `Config -> Config` to create an opaque channel.

```
newchannelWith :: (Config -> Config) -> Channel
```

And `combineWith` would then be etc.

```
combineWith :: Channel -> Stream m a -> Stream m a -> Stream m a
```

Or use `toChannel` and `fromChannel` accordingly.

Reply:
Or should we just have `Channel` instead of `Config -> Config` everywhere?

Reply:
Binary ops should always take in a `Channel`.